src/Types/InvalidTypeException.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Types;
+
+use InvalidArgumentException;
+
+use function sprintf;
+
+final class InvalidTypeException extends InvalidArgumentException
+{
+    public static function invalidTypeName(string $name): self
```

Comment:
The factory methods of exceptions are internal.
```suggestion
    /** @internal */
    public static function invalidTypeName(string $name): self
```